mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergInputFormat.java

```diff
@@ -155,11 +157,31 @@ public void testJoinTables() {
     Assert.assertArrayEquals(new Object[] {1L, "Bob", 102L, 33.33d}, rows.get(2));
   }
 
-  private void createHiveTable(String table, String location) {
+  protected void createTable(String tableName, Schema schema, List<Record> records)
+          throws IOException {
+    Table table = createIcebergTable(tableName, schema, records);
+    createHiveTable(tableName, table.location());
+  }
+
+  protected Table createIcebergTable(String tableName, Schema schema, List<Record> records)
+          throws IOException {
+    String identifier = testTables.identifier("default." + tableName);
+    TestHelper helper = new TestHelper(
+            metastore.hiveConf(), testTables.tables(), identifier, schema, SPEC, FileFormat.PARQUET, temp);
+    Table table = helper.createTable();
+
+    if (!records.isEmpty()) {
+      helper.appendToTable(helper.writeFile(null, records));
+    }
+
+    return table;
+  }
+
+  protected void createHiveTable(String tableName, String location) {
     shell.execute(String.format(
```

Comment:
shouldn't we set the  custom catalog loader  by setting a table property `catalog.loader.class` here?

Reply:
I think if we make `createHiveTable` [and maybe `createIcebergTable`] abstract, each implementation can set the right configuration in table properties. We can then test custom catalog and other configs without explicitly passing as properties from `TestTables` . What do you think?

Reply:
If that works [maybe we also have to abstract `Configuration creation`] then I think we can drop the `TestTables` class as well

Reply:
Yeah, I think that will work once the catalog loader handles catalog settings defined at the table properties level. I'll add this refactor in the follow up PR.